tools/test_api.py

```diff
@@ -2274,20 +2274,21 @@ def test_spec_from_test_builds(test_builds):
     for build in test_builds:
         # Convert TZ target name to test spec platform name
         #
-        # 1. All TZ targets should have name pattern: PLATFORM_[PSA_]S/NS, where:
+        # 1. All TZ targets should have name pattern: PLATFORM_[PSA_/NPSA_]S/NS, where:
         #    (1) 'PLATFORM' for test spec platform name
-        #    (2) 'PSA' is optional to distinguish PSA/non-PSA targets, especially when
-        #        both PSA/non-PSA targets are supported
+        #    (2) 'PSA/NPSA' for PSA/non-PSA targets. Defaults to PSA target on absent.
         #    (3) 'S'/'NS' for secure/non-secure targets
         # 2. Secure target may participate in Greentea, so its name is also truncated here.
         if Target.get_target(test_builds[build]['platform']).is_TrustZone_target:
             if test_builds[build]['platform'].endswith('_NS'):
                 test_builds[build]['platform'] = test_builds[build]['platform'][:-3]
             elif test_builds[build]['platform'].endswith('_S'):
                 test_builds[build]['platform'] = test_builds[build]['platform'][:-2]
-            
+
             if test_builds[build]['platform'].endswith('_PSA'):
```

Comment:
@jamesbeyond  Since we do not have any PSA targets that ends with `_PSA` can we remove this check?

Reply:
Remove **PSA** check or keep it (no change)?

Reply:
Possibly to remote, otherwise not blocking this PR. @ccli8 rather to check the above comment

Reply:
@0xc0170 @devran01 @jamesbeyond Removed **PSA** check from test spec name conversion